gui/ui/src/pages/datasets/medical.folder.dataset/MedicalFolderMetaData.js

```diff
@@ -31,7 +31,7 @@ const MedicalFolderMetadata = (props) => {
                             name={"tags"}
                             type="text"
                             placeholder="Enter tags"
-                            onChange={(e) => { setMedicalFolderDatasetMetadata('tags', e.target.value)}}
+                            onChange={(e) => { setMedicalFolderDatasetMetadata('tags', e.target.value.map(x => {return x.trim()})); }}
```

Comment:
Hi @ybouilla,

Thank you for working on this bug. I think this problem has to be solved in `Tag` component to apply it in all the dataset types that it is used, rather than in `MedicalFolderMetaData`.

Reply:
Thanks, I missed this part. The bug was also existing with common dataset. I extended the fix to every type of data we are currently supportng